packages/block-library/src/utils/interactivity/directives.js

```diff
@@ -51,6 +53,28 @@ export default () => {
 		}
 	);
 
+	directive(
+		'portal',
+		( {
+			directives: {
+				portal: { default: portal },
+			},
+			props: { children },
+			context: inherited,
+		} ) => {
+			const { Provider } = inherited;
+			const inheritedValue = useContext( inherited );
+			return (
+				<>
+					{ createPortal(
+						<Provider value={inheritedValue}>{children}</Provider>,
+						document.body
+					) }
+				</>
```

Comment:
it seems like we could eliminate the fragment `<>` here since `createPortal()` isn't JSX anyway, unless I'm missing something and it somehow returns a list of things that's not a list?

```js
return createPortal(
	<Provider value={inheritedValue}>{children}</Provider>,
	document.body
);
```